attack.go

```diff
@@ -111,10 +113,20 @@ func attack(opts *attackOpts) (err error) {
 		src = files[opts.targetsf]
 		hdr = opts.headers.Header
 	)
-	if opts.lazy {
-		tr = vegeta.NewLazyTargeter(src, body, hdr)
-	} else if tr, err = vegeta.NewEagerTargeter(src, body, hdr); err != nil {
-		return err
+
+	switch opts.format {
+	case "json":
+		tr = vegeta.NewJSONTargeter(src, body, hdr)
+	case "legacy":
+		fallthrough
+	default:
```

Comment:
The user should get feedback for wrong inputs, would move this up to the case where it belongs and error for everything else.

Reply:
Yep, this implementation was rather incomplete in view of getting earlier feedback. I'll add proper validation as well as tests.